Bot.DSharp/DSharpChannel.cs

```diff
@@ -25,6 +25,14 @@ public DSharpChannel(DiscordChannel wrapped)
 
 		public string Name => Wrapped.Name;
 
+		public async Task ClearOverwrites()
+        {
+			foreach(var o in Wrapped.PermissionOverwrites)
```

Comment:
The way this is written is theoretically slower than it could be. It could instead be a non-async task, and be like so:

var tasks = Wrapped.Overwrites.Select(o => o.DeleteAsync()).ToArray():
return Task.WhenAll(tasks);

This would allow multiple slow async operations to kick off at once, rather than waiting for each to complete before moving on to the next one.

Reply:
This is where I'm not really familiar with the Task APIs particularly. I can try this though. This would still be an async Task function though from what I can tell.